targets/TARGET_STM/serial_api.c

```diff
@@ -0,0 +1,122 @@
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *******************************************************************************
+ */
+#include "mbed_assert.h"
+#include "mbed_error.h"
+#include "serial_api.h"
+#include "serial_api_hal.h"
+
+#if DEVICE_SERIAL
+
+void init_uart(serial_t *obj)
+{
+    struct serial_s *obj_s = SERIAL_S(obj);
+    UART_HandleTypeDef *huart = &uart_handlers[obj_s->index];
+    huart->Instance = (USART_TypeDef *)(obj_s->uart);
+
+    huart->Init.BaudRate     = obj_s->baudrate;
+    huart->Init.WordLength   = obj_s->databits;
+    huart->Init.StopBits     = obj_s->stopbits;
+    huart->Init.Parity       = obj_s->parity;
+#if DEVICE_SERIAL_FC
+    huart->Init.HwFlowCtl    = obj_s->hw_flow_ctl;
+#else
+    huart->Init.HwFlowCtl    = UART_HWCONTROL_NONE;
+#endif
+    huart->Init.OverSampling = UART_OVERSAMPLING_16;
+    huart->TxXferCount       = 0;
+    huart->TxXferSize        = 0;
+    huart->RxXferCount       = 0;
+    huart->RxXferSize        = 0;
+
+    if (obj_s->pin_rx == NC) {
+        huart->Init.Mode = UART_MODE_TX;
+    } else if (obj_s->pin_tx == NC) {
+        huart->Init.Mode = UART_MODE_RX;
+    } else {
+        huart->Init.Mode = UART_MODE_TX_RX;
+    }
+
+    if (HAL_UART_Init(huart) != HAL_OK) {
+        error("Cannot initialize UART\n");
+    }
+}
+
+void serial_format(serial_t *obj, int data_bits, SerialParity parity, int stop_bits)
+{
+    struct serial_s *obj_s = SERIAL_S(obj);
+
+    switch (parity) {
+        case ParityOdd:
+            obj_s->parity = UART_PARITY_ODD;
+            break;
+        case ParityEven:
+            obj_s->parity = UART_PARITY_EVEN;
+            break;
+        default: // ParityNone
+        case ParityForced0: // unsupported!
+        case ParityForced1: // unsupported!
+            obj_s->parity = UART_PARITY_NONE;
+            break;
+    }
+
+    switch (data_bits) {
+        case 9:
+            MBED_ASSERT(parity == UART_PARITY_NONE);
+            obj_s->databits = UART_WORDLENGTH_9B;
+            break;
+        default:
+        case 8:
+            if (parity != UART_PARITY_NONE) {
+                obj_s->databits = UART_WORDLENGTH_9B;
+            } else {
+                obj_s->databits = UART_WORDLENGTH_8B;
+            }
+            break;
+#if defined UART_WORDLENGTH_7B
+        case 7:
+            if (parity != UART_PARITY_NONE) {
+                obj_s->databits = UART_WORDLENGTH_8B;
+            } else {
+                obj_s->databits = UART_WORDLENGTH_7B;
+            }
+            break;
+#endif
+    }
+
+    if (stop_bits == 2) {
+        obj_s->stopbits = UART_STOPBITS_2;
+    } else {
+        obj_s->stopbits = UART_STOPBITS_1;
+    }
+
+    init_uart(obj);
+}
+
+#endif /* DEVICE_SERIAL */
```

Comment:
I think the github viewer might be ignoring the last line. If I click edit this file I can see line 64 being a blank line. Also on the file information on top it says "64 lines (53 sloc)" but on the viewer lines are only numbered 1 to 63. Have you checked out the change and see if the line is actually there? I can't do it now at work otherwise it'll have to wait until I have a moment at home. But I think the line is there